resources/views/partials/talk-in-list.blade.php

```diff
@@ -1,12 +1,25 @@
 <div class="pull-right">
+  <a href="{{ route('talks.edit', ['id' => $talk->id]) }}" class="btn btn-xs btn-primary">
+      <span class="glyphicon glyphicon-edit" aria-hidden="true"></span>
+      Edit
+  </a>
     <a href="{{ route('talks.delete', ['id' => $talk->id]) }}" class="btn btn-xs btn-danger">
         <span class="glyphicon glyphicon-trash" aria-hidden="true"></span>
         Delete
     </a>
-    <a href="{{ route('talks.edit', ['id' => $talk->id]) }}" class="btn btn-xs btn-primary">
-        <span class="glyphicon glyphicon-edit" aria-hidden="true"></span>
-        Edit
-    </a>
+    @if (Route::is('talks.archive'))
```

Comment:
I think you'd do better to switch this based on a property of the talk itself; e.g. `$talks->isArchived()`